pkg/test/framework/components/namespace/kube.go

```diff
@@ -76,6 +76,17 @@ func (n *kubeNamespace) Dump() {
 				if err = ioutil.WriteFile(fname, []byte(l), os.ModePerm); err != nil {
 					scopes.CI.Errorf("Unable to write logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
 				}
+
+				if container.Name == "istio-proxy" {
+					if cfgDump, err := cluster.Exec(pod.Namespace, pod.Name, container.Name, "curl http://127.0.0.1:15000/config_dump"); err == nil {
```

Comment:
This won't work on distroless because `curl` is not present. I would suggest we either add

* Call to sidecar.Config() in https://github.com/istio/istio/blob/c11c7c25cdead0b535e103bd206ab76bd4d21ad3/pkg/test/framework/components/echo/kube/workload.go#L108. Downside is this only applies to `echo` not a generic pod
* Switch to `pilot-agent request GET config_dump` which will work on distroless